platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/network/protocol/EmergencyReconnectProtocolFactory.java

```diff
@@ -0,0 +1,117 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.swirlds.platform.network.protocol;
+
+import com.swirlds.base.time.Time;
+import com.swirlds.common.context.PlatformContext;
+import com.swirlds.common.notification.NotificationEngine;
+import com.swirlds.common.platform.NodeId;
+import com.swirlds.common.threading.manager.ThreadManager;
+import com.swirlds.config.api.Configuration;
+import com.swirlds.platform.metrics.ReconnectMetrics;
+import com.swirlds.platform.reconnect.ReconnectController;
+import com.swirlds.platform.reconnect.ReconnectThrottle;
+import com.swirlds.platform.reconnect.emergency.EmergencyReconnectProtocol;
+import com.swirlds.platform.recovery.EmergencyRecoveryManager;
+import com.swirlds.platform.state.signed.ReservedSignedState;
+import com.swirlds.platform.system.status.StatusActionSubmitter;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.time.Duration;
+import java.util.Objects;
+import java.util.function.Supplier;
+/**
+ * Implementation of a protocol factory for emergency reconnect
+ */
+public class EmergencyReconnectProtocolFactory implements ProtocolFactory {
+
+    private final EmergencyRecoveryManager emergencyRecoveryManager;
+    private final ReconnectThrottle teacherThrottle;
+    private final Supplier<ReservedSignedState> emergencyStateSupplier;
+    private final Duration reconnectSocketTimeout;
+    private final ReconnectMetrics reconnectMetrics;
+    private final ReconnectController reconnectController;
+    private final ThreadManager threadManager;
+    private final NotificationEngine notificationEngine;
+    private final Configuration configuration;
+    private final Time time;
+    private final PlatformContext platformContext;
+    private final StatusActionSubmitter statusActionSubmitter;
+
+    /**
+     * @param  platformContext         the platform context
+     * @param time                     provides wall clock time
+     * @param threadManager            responsible for managing thread lifecycles
+     * @param notificationEngine       the notification engine to use
+     * @param emergencyRecoveryManager the state of emergency recovery, if any
+     * @param teacherThrottle          restricts reconnects as a teacher
+     * @param emergencyStateSupplier   returns the emergency state if available
+     * @param reconnectSocketTimeout   the socket timeout to use when executing a reconnect
+     * @param reconnectMetrics         tracks reconnect metrics
+     * @param reconnectController      controls reconnecting as a learner
+     * @param statusActionSubmitter    enables submitting platform status actions
+     * @param configuration            the platform configuration
+     */
+    public EmergencyReconnectProtocolFactory(
+            @NonNull final PlatformContext platformContext,
+            @NonNull final Time time,
```

Comment:
You can remove this `Time` argument, and instead use `platformContext.getTime()`. Time didn't used to be in the platform context, but now that it is we should try to avoid passing Time separately.